docs/vNext/DeclarativeWorkloadModule.md

```diff
@@ -0,0 +1,190 @@
+
+### Contracts
+
+Used to specify the list of contracts to be tested. Each `contracts` list element has the following format.
+
+|Property|Type       |Description|
+|:-------|:----------|:----------|
+|name    |string     |Name of SUT contract to be tested|
+|functionSelection|string|Type of contract picking logic|
+|function|list|List of Contract functions|
+
+### Functions
+
+Used to specify the list of functions under a contract to be tested. Each `functions` list element has the following format.
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|name    |string|Name of SUT function to be tested|
+|parameters|list|List of user-defined parameters to be generated|
+
+### Parameters
+
+Used to specify different generated parameters for each function. The `parameters` list can contain one or more of the following items.
+
+#### Uniform Random
+
+Value provider format for generating a random number within a given range.
+
+##### Example
+
+```yaml
+- name: randomNumber
+  type: uniform_random
+  options:
+    min: 0
+    max: 100
+```
+##### Attributes
+
+|Property   |Type  |Description                                        |
+|:----------|:-----|:--------------------------------------------------|
+|type       |string|Assigned the value `uniform_random`                |
+|name       |string|Parameter Name                                     |
+|options.min|number|Mininum inclusive range for generated random number|
+|options.max|number|Maximum inclusive range for generated random number|
+
+#### Parameter Reference
+
+Value Provider format for referencing a `module.arguments` item.
+
+##### Example
+```yaml
+- name: referenceNumber
+  type: parameter_reference
+  options:
+    name: marbleNumber #assuming this is defined under round.arguments
+```
+##### Attributes
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|type    |string|Assigned the value `parameter_reference`|
+|name    |string|Parameter name|
+|options.name|string|Should be a variable name specified under `module.arguments`|
+
+#### Variable Reference
+
+Value Provider format for referencing a Caliper workload variable.
+
+##### Example
+```yaml
+- name: referenceNumber
+  type: variable_reference
+  options:
+    name: txIndex #a Caliper workload parameter
+```
+##### Attributes
+
+|Property|Type|Description|
+|:-------|:---|:----------|
+|type    |string|Assigned the value `variable_reference`.|
+|name    |string|Parameter name|
+|options.name    |string|Parameter Name. Should refer to a Caliper variable. |
+
+#### List Element
+
+Value provider format for selecting an item from a given list.
+##### Example
+
+```yaml
+- name: selectedColor
+  type: list_element
+  options:
+    list: ['red', blue', 'green'],
+    selector:
+      type: parameter_reference
```

Comment:
A variable reference to the txIndex variable would be a better example, as it demonstrates a round-robin selection